pkg/prometheus/operator.go

```diff
@@ -1428,6 +1601,56 @@ func (c *Operator) selectServiceMonitors(p *monitoringv1.Prometheus) (map[string
 		})
 	}
 
+	// If denied by Prometheus spec, filter out all service monitors that access
+	// the file system.
+	if p.Spec.ArbitraryFSAccessThroughSMs.Deny {
+		for namespaceAndName, sm := range res {
+			for _, endpoint := range sm.Spec.Endpoints {
+				if endpoint.BearerTokenFile != "" {
+					delete(res, namespaceAndName)
+					level.Warn(c.logger).Log(
+						"msg", fmt.Sprintf(
```

Comment:
Having printf in structured logging is an "anti pattern". Mabye we can rephrase and give the ServiceMonitor name as a key value pair